include/aligator/core/traj-opt-problem.hxx

```diff
@@ -55,10 +55,19 @@ TrajOptProblemTpl<Scalar>::TrajOptProblemTpl(const ConstVectorRef &x0,
                                              shared_ptr<CostAbstract> term_cost)
     : TrajOptProblemTpl(createStateError(x0, space, nu), term_cost) {}
 
+template <typename Scalar> void TrajOptProblemTpl<Scalar>::configure() const {
+  /// TODO configure initial constraints and final cost ?
```

Comment:
@ManifoldFR Do you think we must manage common data for initial constraint and final cost/constraints ?

Reply:
One possibility would be to create a dummy, no-dynamics `StageModel` ? This is what Crocoddyl has done for years to handle the cost function by using a terminal `croccodyl::ActionModel` : they do call into their dynamics at the last stage however, but we don't _have_ to

Reply:
StageModel store some Manifold that are initialized from a Dynamics. So we should add a constructor that will allow to hack this.
Also, when modifying StageModel, we have to be aware that it can also hold initial and final constraints.

I prefer to create a new data structure that will store Cost, Constraints and CommonModel.
This structure will be used by StageModel so the code will be well factorized.

What do you think ?
Some name idea : `CostConstraintStack`, `CostConstraintContainer`